integration-tests/plugins/__tests__/inventory/variant-inventory-service.js

```diff
@@ -0,0 +1,273 @@
+const path = require("path")
+
+const { bootstrapApp } = require("../../../helpers/bootstrap-app")
+const { initDb, useDb } = require("../../../helpers/use-db")
+const {
+  simpleProductVariantFactory,
+  simpleProductFactory,
+} = require("../../factories")
+
+jest.setTimeout(50000)
+
+describe("Inventory Module", () => {
+  let appContainer
+  let dbConnection
+  let express
+
+  let invItem1
+  let invItem2
+  let variant1
+  let variant2
+
+  beforeAll(async () => {
+    const cwd = path.resolve(path.join(__dirname, "..", ".."))
+    dbConnection = await initDb({ cwd })
+    const { container, app, port } = await bootstrapApp({ cwd, verbose: false })
+    appContainer = container
+
+    express = app.listen(port, (err) => {
+      process.send(port)
+    })
+  })
+
+  afterAll(async () => {
+    const db = useDb()
+    await db.shutdown()
+    express.close()
+  })
+
+  describe("ProductVariantInventoryItem", () => {
+    afterEach(async () => {
+      const db = useDb()
+      return await db.teardown()
+    })
+
+    beforeEach(async () => {
+      const inventoryService = appContainer.resolve("inventoryService")
+
+      const { variants } = await simpleProductFactory(dbConnection, {
+        variants: [{}, {}],
+      })
+
+      variant1 = variants[0]
+      variant2 = variants[1]
+
+      invItem1 = await inventoryService.createInventoryItem({
+        sku: "test-sku-1",
+      })
+
+      invItem2 = await inventoryService.createInventoryItem({
+        sku: "test-sku-2",
+      })
+    })
+
+    it("Attaches a single item with spread params", async () => {
+      const pviService = appContainer.resolve("productVariantInventoryService")
+      await pviService.attachInventoryItem(variant1.id, invItem1.id)
+
+      const variantItems = await pviService.listByVariant(variant1.id)
+      expect(variantItems.length).toEqual(1)
+      expect(variantItems[0]).toEqual(
+        expect.objectContaining({
+          inventory_item_id: invItem1.id,
+          variant_id: variant1.id,
+        })
+      )
+    })
+
+    it("Attaches a multiple inventory items and variants at once", async () => {
+      const pviService = appContainer.resolve("productVariantInventoryService")
+      await pviService.attachInventoryItem([
```

Comment:
They could be plural but that would break the entire api of the service. I've kept them singular for that reason. I can extend the api if we want to do that instead to have two methods, one singular to create single entities and one bulk

Reply:
yes this is my suggestion to have both and the singular that calls the plural :) sorry for the miss understanding :) since bulk can also create only one :)